Add remote-protocol at TcpClients

openBackhaul opened the pull request:
Find list of changes in issue #344

Fixes #344
Fixes #355
Fixes #356

PrathibaJee reviewed (commented):
.

PrathibaJee reviewed (changes requested):
> Couldnt able to comment in the specific lines , thats why providing it as a common comment.

- **ApplicationPattern.yaml :**
      Overall , the  file looks good. All the services got complemented with an attribute for protocol with proper reference to the tcp-server/client protocol, description and examples.
      _**Comment :**_ 
      for all the callbacks , currently , the protocols are starting with https://
      https://[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp.......
      Is this is fine or this needs to be a reference to the remote-protocol? Kindly let me know.

- **ApplicationPattern+data.json**
      _**Comment :**_ 
      In this file , in all the action-profile-configuration , require to change the protocol-reference from
      /tcp-server-interface-configuration/protocol
      to
      /tcp-server-interface-configuration/local-protocol

- **ApplicationPattern+profileInstances.yaml :**
      _**Comment :**_ 
      In this file , for the protocol-reference attribute ofthe action profiles *-action-p-000, *-action-p-001, *-action-p-002 require to modify the protocol-reference to point to the "local-protocol"

- **ApplicationPattern+services.yaml**
      _**Comment :**_ 
      In this file , whether we have to include the protocol attribute for the tcp-clients too ? Kindly let me know. Thanks.

openBackhaul commented on ApplicationPattern.yaml:
```diff
@@ -7984,17 +8107,17 @@ paths:
                   enum:
                     - 'tcp-server-interface-1-0:PROTOCOL_TYPE_HTTP'
                     - 'tcp-server-interface-1-0:PROTOCOL_TYPE_HTTPS'
                     - 'tcp-server-interface-1-0:PROTOCOL_TYPE_NOT_YET_DEFINED'
```
Same reason as in 5G-xhaul. If device does not support HTTP or HTTPS the management interface has to represent the default value. Just ignore it.

openBackhaul commented:
In OAS at callbacks:
Replaced _https_ by _[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp-client-interface-1-0:tcp-client-interface-pac/tcp-client-interface-configuration/remote-protocol]_

In CONFIGfile:
Changed from _/tcp-server-interface-configuration/protocol_ to _/tcp-server-interface-configuration/local-protocol_

In ProfileInstanceList:
Changed from _/tcp-server-interface-configuration/protocol_ to _/tcp-server-interface-configuration/local-protocol_

In ServiceList:
_protocol_ attribute added to TcpClients

IswaryaaS commented on ApplicationPattern.yaml:
```diff
@@ -600,6 +636,7 @@ paths:
                         - application-release-number
                         - embedding-operation
                         - client-update-operation
+                        - application-protocol
```
line 620 : callback uri definition for PromptForRegisteringCausesRegistrationRequest has protocol https which needs to be updated to reference registry-office-protocol 
{$request.body#registry-office-protocol}://...

IswaryaaS commented on ApplicationPattern.yaml:
```diff
@@ -600,6 +636,7 @@ paths:
                         - application-release-number
                         - embedding-operation
                         - client-update-operation
+                        - application-protocol
```
line 620 : callback uri definition for PromptForRegisteringCausesRegistrationRequest has protocol https which needs to be updated to reference registry-office-protocol 
{$request.body#registry-office-protocol}://...

openBackhaul replied:
Corrected as proposed. Many thanks for indicating.

IswaryaaS commented on ApplicationPattern.yaml:
```diff
@@ -1067,6 +1146,7 @@ paths:
                 service-log-application: 'ExecutionAndTraceLog'
                 service-log-application-release-number: '2.0.0'
                 service-log-operation: '/v1/record-service-request'
+                service-log-protocol: 'HTTP'
```
line 1201: callback uri definition for ServiceRequestCausesLoggingRequest has protocol https which needs to be updated to reference service-log-protocol 
{$request.body#service-log-protocol}://...

IswaryaaS commented on ApplicationPattern.yaml:
```diff
@@ -1308,6 +1397,7 @@ paths:
                 oam-log-application: 'OamLog'
                 oam-log-application-release-number: '2.0.0'
                 oam-log-operation: '/v1/record-oam-request'
+                oam-log-protocol: 'HTTP'
```
line 1452: callback uri definition for OamRequestCausesLoggingRequest has protocol https which needs to be updated to reference oam-log-protocol {$request.body#oam-log-protocol}://...

openBackhaul replied:
Changed as proposed. Many thanks for the concrete proposal.

IswaryaaS commented on ApplicationPattern.yaml:
```diff
@@ -1632,6 +1731,7 @@ paths:
                 oam-approval-application: 'AdministratorAdministration'
                 oam-approval-application-release-number: '2.0.0'
                 oam-approval-operation: '/v1/approve-oam-request'
+                oam-approval-protocol: 'HTTP'
```
line 1786: callback uri definition for OamRequestCausesInquiryForAuthentication has protocol https which needs to be updated to reference oam-approval-protocol {$request.body#oam-approval-protocol}://...

openBackhaul replied:
Changed as proposed. Many thanks for the concrete proposal.

IswaryaaS commented on ApplicationPattern.yaml:
```diff
@@ -2191,6 +2310,7 @@ paths:
                 topology-operation-fc-update: '/v1/update-fc'
                 topology-operation-fc-port-update: '/v1/update-fc-port'
                 topology-operation-fc-port-deletion: '/v1/delete-fc-port'
+                topology-application-protocol: 'HTTP'
```
uri definitions for callbacks of /v1/redirect-topology-change-information has protocol https which needs to be updated to reference topology-application-protocol {$request.body#topology-application-protocol}://...

openBackhaul commented:
After replacing _https_ by _[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp-client-interface-1-0:tcp-client-interface-pac/tcp-client-interface-configuration/remote-protocol], the YAML didn't successfully validate in Postman API editor anymore.
Decided to substitute _https_ by 
````
url: #[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp-client-interface-1-0:tcp-client-interface-pac/tcp-client-interface-configuration/remote-protocol]://[...]
````

IswaryaaS commented on ApplicationPattern+data.json:
```diff
@@ -23,7 +23,7 @@
                 "action-profile-configuration": {
                   "request": {
-                    "protocol-reference": "/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-s-bm-001/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/protocol",
+                    "protocol-reference": "/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-s-bm-001/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol",
```
we have uuid in profile section starting with prefix * and xx, which will be replaced by the actual application-name's notation in individual application. Here, uuid prefix is * in profile section whereas xx in logical-termination-point list. please let me know if we can maintain a common pattern for denoting the application-name holder (either * or xx) to maintain hormony.

openBackhaul replied:
Made a CTRL+h from *-1-0-0 to xx-1-0-0. Thanks for indicating.
